### PR TITLE
Mason Edits GPU on Yens Blog Post

### PR DESCRIPTION
**Pull request by mpj2104**
`blog/2024/04/25/run-jobs-on-yens-gpu-node/`

### CODE REVIEW

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 1]
```diff
@@ -11,69 +11,60 @@ subtitle: Running Jobs on Yen's GPU node
 
 ---
 
-# Run Jobs on Yen's GPU node
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:14]: the author was me :)

Comment (natalya-patrikeeva) [reply docs/blog/posts/2024-04-25-yen_gpu.md:14]: We also decided not to have multiple dates (so remove updated)

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 2]
```diff
@@ -11,69 +11,60 @@ subtitle: Running Jobs on Yen's GPU node
...
+# Run Jobs on Yen's GPU Node
 
 ## GPU Node Overview
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:16]: GPU partition is more accurate since it's not just one node

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 3]
```diff
@@ -11,69 +11,60 @@ subtitle: Running Jobs on Yen's GPU node
...
-The Yen-slurm has three GPU nodes, `yen-gpu1` node with 64 threads, 256 G of RAM and 4 A30 NVIDIA GPU's, `yen-gpu2` and `yen-gpu3` nodes -- both with 64 threads, 256 G of RAM and 4 A40 NVIDIA GPU's.
-The A30 NVIDIA GPU's have 24 G of GPU RAM while the A40 NVIDIA GPU's have 48 G of GPU RAM per GPU.
+[Yen Slurm](/_user_guide/slurm/){:target="_blank"} has three GPU nodes: 
 
+- `yen-gpu1` has 64 threads, 256 G of RAM and 4 A30 NVIDIA GPU's
+- `yen-gpu2` and `yen-gpu3` each have 64 threads, 256 G of RAM and 4 A40 NVIDIA GPU's
 
-<!-- more -->
-## Login to Yens
+The A30 NVIDIA GPU's have 24 G of GPU RAM while the A40 NVIDIA GPU's have 48 G of GPU RAM.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:22]: I would add the more separator here so only the preview shows up

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 4]
```diff
@@ -11,69 +11,60 @@ subtitle: Running Jobs on Yen's GPU node
...
-This example uses the MNIST dataset for image classification, and consists of a simple fully connected neural network 
-with one hidden layer. 
+#### Create Python Script
+This example uses the MNIST dataset for image classification, and consists of a simple fully connected neural network with one hidden layer. 
 
-Save the following code to a new file called `mnist.py`.
+Save the following code to a new file called `mnist.py`:
 
-```python
+```python title="mnist.py"
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:67]: can we add line numbers?

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 5]
```diff
@@ -11,69 +11,60 @@ subtitle: Running Jobs on Yen's GPU node
...
-Python 3.10.12
+```{.yaml .no-copy title="Terminal Output"}
+...
+nvidia-cuda-runtime-cu12  12.1.105
+...
+pytorch-lightning         2.1.3
+...
 ```
 
-If you need additional packages installed, you can `pip install` them to your `~/.local` since this global `venv`
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:67]: is this what we recommend?

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 6]
```diff
@@ -140,12 +131,11 @@ with torch.no_grad():
...
 
+#### Submit Script to Yen Slurm
 
-### Submit Slurm script
+You can now write a Slurm submission script to run `mnist.py` on a GPU node on `yen-slurm`. An example submission script `train-gpu.slurm` looks like:
 
-The submission script `train-gpu.slurm` looks like:
-
-```bash
+```slurm title="train-gpu.slurm"
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:138]: the limit is now 4 GPU's per user

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 7]
```diff
@@ -261,69 +249,56 @@ Accuracy on the test set: 97 %
...
-$ python -m ipykernel install --user --name pytorch201 --display-name 'PyTorch 2.0.1'
+To add the PyTorch environment, load the `pytorch` module and then create a JupyterHub kernel from the `venv` using the following commands:
+```bash title="Terminal Command"
+ml pytorch
+python -m ipykernel install --user --name pytorch201 --display-name 'PyTorch 2.0.1'
 ```
 
-Launch <a href="/yen/webBasedCompute.html" target="_blank">JupyterHub</a> on the *same* interactive yen that you have
-loaded the `pytorch` module. You should see a new `PyTorch 2.0.1` Notebook kernel you can start up.
+Now launch JupyterHub on the *same* interactive yen that you have loaded the `pytorch` module on. When you launch a new Python notebook, you should see the just created `PyTorch 2.0.1` notebook kernel that you can start up.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:267]: i think we can use the kernel on any node now. Don't need to start it on the same yen

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 8]
```diff
@@ -261,69 +249,56 @@ Accuracy on the test set: 97 %
...
 
 !!! warning
-    You have to load `pytorch` module in the terminal before you can launch the `PyTorch 2.0.1` kernel. If you do not load the module in the terminal, `torch` will not import in the notebook.
-
-Once you start up the notebook, make sure you can import `torch` but CUDA is not available (since interactive yens do
-not have GPU's).
+    You have to load the `pytorch` module in the terminal before you can launch the `PyTorch 2.0.1` kernel. If you do not load the module in the terminal, `torch` will not import in the notebook.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:270]: i don't think we have to do that anymore

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 9]
```diff
@@ -261,69 +249,56 @@ Accuracy on the test set: 97 %
...
-Python 3.10.11
+```{.yaml .no-copy title="Terminal Output"}
+...
+base                  *  /software/free/tensorflow/2
+...
 ```
 
-If you need additional packages installed, you can `pip install` them to your `~/.local` since this global conda environment
-is not user writable. 
+If you need additional packages installed, you will need to `pip install` them to your `~/.local` since this global conda environment is not user writable.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:294]: again, is that recommended?

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 10]
```diff
@@ -380,11 +355,11 @@ print("Test loss:", score[0])
...
+#### Submit Script to Yen Slurm
 
-The submission script `train-gpu.slurm` looks like:
+You can now write a Slurm submission script to run `mnist-keras.py` on a GPU node on `yen-slurm`. An example submission script `train-gpu.slurm` looks like:
 
-```bash
+```slurm title="train-gpu.slurm"
 #!/bin/bash
 
 # Example slurm script to train keras DL model on Yen GPU
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:365]: the limit is 4 GPUs per user

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 11]
```diff
@@ -413,45 +388,42 @@ source activate
 python mnist-keras.py
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:388]: let's add line numbers as this script is long

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 12]
```diff
@@ -503,27 +475,27 @@ Test loss: 0.02380027435719967
...
 
-```bash
-$ module load tensorflow
-$ source activate
-$ python -m ipykernel install --user --name tf2 --display-name 'Tensorflow 2.11'
+To add the Keras environment, load the `tensorflow` module and then create a JupyterHub kernel from the conda environment using the following commands:
+```bash title="Terminal Command"
+module load tensorflow
+source activate
+python -m ipykernel install --user --name pytorctf2h201 --display-name 'Tensorflow 2.11'
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:491]: what is pytorctf2h201?

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 13]
```diff
@@ -503,27 +475,27 @@ Test loss: 0.02380027435719967
...
 ```
 
-Launch <a href="/yen/webBasedCompute.html" target="_blank">JupyterHub</a> on the *same* interactive yen that you have 
-loaded the `tensorflow` module. You should see a new `Tensorflow 2.11` Notebook kernel you can start up. 
+Now launch JupyterHub on the *same* interactive yen that you have loaded the `tensorflow` module on. When you launch a new Python notebook, you should see the just created `Tensorflow 2.11` notebook kernel that you can start up.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:494]: don't need this

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 14]
```diff
@@ -503,27 +475,27 @@ Test loss: 0.02380027435719967
...
 
 !!! warning
-    You have to load `tensorflow` module in the terminal before you can launch the `Tensorflow 2.11` kernel. If you do not load the module in the terminal, `tensorflow` will not import in the notebook.
+    You have to load the `tensorflow` module in the terminal before you can launch the `Tensorflow 2.11` kernel. If you do not load the module in the terminal, `tensorflow` will not import in the notebook.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:497]: don't need this anymore either

**Review by natalya-patrikeeva — CHANGES_REQUESTED** [review]
I don't know if it's possible to combine the scripts and just have 2 tabs for the python scripts. because the rest of instructions seem duplicated for pytorch vs keras

File: docs/blog/posts/2024-04-25-yen_gpu.md (hunk 5 above)
Comment (mpj2104) [reply docs/blog/posts/2024-04-25-yen_gpu.md:67]: hmmmm, i dont know what is the best practice here besides creating your own copy of the environment, perhaps i will remove it for now and we can make another blog post about it in the future when we know whats best

**Comment by mpj2104** [comment]
> I don't know if it's possible to combine the scripts and just have 2 tabs for the python scripts. because the rest of instructions seem duplicated for pytorch vs keras

Great idea, makes the page much cleaner. I did that in my latest commit to this branch.

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 15]
```diff
@@ -1,210 +1,281 @@
...
-$ sinfo -p gpu
+```{.yaml .no-copy title="Terminal Output"}
 PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
-gpu          up 1-00:00:00      2   idle yen-gpu[1-3]
+gpu          up 1-00:00:00      1    mix yen-gpu3
+gpu          up 1-00:00:00      2   idle yen-gpu[1-2]
 ```
 
 !!! warning
     There is a limit of 1 day runtime and 2 GPU's per user.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:36]: actually this is now 4 GPU's per user

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 16]
```diff
@@ -241,289 +312,64 @@ Tue May  9 15:58:57 2023
...
+    source activate
+    python -m ipykernel install --user --name tensorflow211 --display-name 'Tensorflow 2.11'
+    ```
+
+When you launch a new Python notebook, you should see the just created `PyTorch 2.0.1` or `Tensorflow 2.11` notebook kernel that you can start up.
+
+Once you start up the notebook, make sure you can import `torch` (for PyTorch) or `tensorflow` (for Keras). Note that CUDA is not available here since the interactive Yens do not have GPUs.
+
+=== "PyTorch"
+    ![](/assets/images/pytorch-kernel.png)
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:373]: per Alex, images need to have alt tags for accessibility reasons

File: docs/blog/posts/2024-04-25-yen_gpu.md [hunk 17]
```diff
@@ -241,289 +312,64 @@ Tue May  9 15:58:57 2023
...
-Once you start up the notebook, make sure you can import `tensorflow` but CUDA is not available (since interactive yens do 
-not have GPU's). 
-
-![](/images/tf-kernel.png)
\ No newline at end of file
+To add the environment, load the module and then create a JupyterHub kernel from the `venv` using the following commands:
+=== "PyTorch"
+    ```bash title="Terminal Command"
+    ml pytorch
+    python -m ipykernel install --user --name pytorch201 --display-name 'PyTorch 2.0.1'
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2024-04-25-yen_gpu.md:359]: I think now the pytorch module loads pytorch 2.1.2

**Review by natalya-patrikeeva — CHANGES_REQUESTED** [review]
super minor edits round 2

**Review by natalya-patrikeeva — APPROVED** [review]
lgtm!!
